integrationtests/all-embedded-query-it/src/test/java/org/infinispan/all/embeddedquery/LocalCacheTest.java

```diff
@@ -0,0 +1,573 @@
+      person2.setAge(10);
+      cache.put(key2, person2);
+
+      {
+         CacheQuery cacheQuery = createCacheQuery(cache, "name", "Goat");
+         List<Object> found = cacheQuery.list();
+
+         assertEquals(2, found.size());
+
+         cacheQuery.sort( sort );
+
+         found = cacheQuery.list();
+
+         assertEquals(2, found.size());
+         assertTrue(found.get(0).equals(person2)); // person2 is 30 and named Goat
+         assertTrue(found.get(1).equals(person3)); // person3 is 25 and named Goat
+      }
+
+      StaticTestingErrorHandler.assertAllGood(cache);
+   }
+
+   @Test
+   public void testSetFilter() {
+      loadTestingData();
+
+      CacheQuery cacheQuery = createCacheQuery(cache, "name", "Goat");
+      List<Object> found = cacheQuery.list();
+
+      assertEquals(2, found.size());
+
+      Filter filter = new PrefixFilter(new Term("blurb", "cheese"));
+
+      cacheQuery.filter(filter);
+
+      found = cacheQuery.list();
+
+      assertEquals(1, found.size());
+      StaticTestingErrorHandler.assertAllGood(cache);
+   }
+
+   @Test
+   public void testLazyIterator() {
+      loadTestingData();
+      CacheQuery cacheQuery = createCacheQuery(cache, "blurb", "playing");
+
+      ResultIterator found = cacheQuery.iterator(new FetchOptions().fetchMode(FetchOptions.FetchMode.LAZY));
+
+      try {
+         assertTrue(found.hasNext());
+         found.next();
+         assertTrue(!found.hasNext());
+      } finally {
+         found.close();
+      }
+      StaticTestingErrorHandler.assertAllGood(cache);
+   }
+
+   @Test(expected = IllegalArgumentException.class)
+   public void testUnknownFetchModeIterator() {
+      loadTestingData();
+      CacheQuery cacheQuery = createCacheQuery(cache, "blurb", "playing");
+
+      ResultIterator found = cacheQuery.iterator(new FetchOptions(){
+         public FetchOptions fetchMode(FetchMode fetchMode) {
+            return null;
+         }
+      });
+
+      try {
+         assertTrue(found.hasNext());
+         found.next();
+         assertTrue(!found.hasNext());
+      } finally {
+         found.close();
+      }
+   }
+
+   @Test
+   public void testIteratorWithDefaultOptions() {
+      loadTestingData();
+      CacheQuery cacheQuery = createCacheQuery(cache, "blurb", "playing");
+
+      ResultIterator found = cacheQuery.iterator();
+
+      try {
+         assertTrue(found.hasNext());
+         found.next();
+         assertTrue(!found.hasNext());
+      } finally {
+         found.close();
+      }
+      StaticTestingErrorHandler.assertAllGood(cache);
+   }
+
+   @Test
+   public void testExplain() {
+      loadTestingData();
+      CacheQuery cacheQuery = createCacheQuery(cache, "blurb", "Eats");
+
+      int matchCounter = 0;
+      int i = 0;
+
+      //The implementation is changed to this way as in case of NRT index manager the number of created documents may
+      //differ comparing to the simple configuration.
+      while (true) {
```

Comment:
Nope, unfortunately there is no other way to query all explanations... We won't get rid of that loop.